### PR TITLE
[CBRD-25210] Add syntax [GRANT|REVOKE] ON {PROCEDURE|FUNCTION}	

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CBRD-25210

The following syntax will be supported.
```
GRANT EXECUTE ON PROCEDURE/FUNCTION <stored procedure/function name> TO <users>
REVOKE EXECUTE ON PROCEDURE/FUNCTION <stored procedure/function name> FROM <users>
```
This PR involves the implementation of front-end modules, query parser, and semantic checker. The following aspects require consideration:

Syntax Check
- Check the above syntax in the query parser.
Semantic Check
- If the designated stored procedure or stored function (<stored procedure/function name>) is not present
- A new semantic error is going to be added
- If the specified users (<users>) do not exist, generate an error (MSGCAT_SEMANTIC_USER_IS_NOT_IN_DB) as it is.

### CODE REVIEW

File: src/compat/dbi.h [hunk 1]
```diff
@@ -148,6 +148,7 @@ extern "C"
   extern int db_drop_member (DB_OBJECT * user, DB_OBJECT * member);
   extern int db_set_password (DB_OBJECT * user, const char *oldpass, const char *newpass);
   extern int db_set_user_comment (DB_OBJECT * user, const char *comment);
+
```
Comment (beyondykk9) [inline src/compat/dbi.h]: it looks useless modification.

File: src/object/authenticate.c [hunk 2]
```diff
@@ -150,36 +151,7 @@ const char *AU_DBA_USER_NAME = "DBA";
...
-         strcmp(name, CT_TRIGGER_NAME) == 0 || \
-         strcmp(name, CT_ROOT_NAME) == 0 || \
-         strcmp(name, CT_PASSWORD_NAME) == 0 || \
-         strcmp(name, CT_AUTHORIZATION_NAME) == 0 || \
-         strcmp(name, CT_AUTHORIZATIONS_NAME) == 0 || \
-	 strcmp(name, CT_CHARSET_NAME) == 0 || \
-         strcmp(name, CT_DB_SERVER_NAME) == 0 || \
-	 strcmp(name, CT_SYNONYM_NAME) == 0 || \
-	 strcmp(name, CT_DUAL_NAME) == 0)
+#define IS_CATALOG_CLASS(name) sm_is_system_class(name)
```
Comment (beyondykk9) [inline src/object/authenticate.c]: it looks not related this issue.

**Comment by hgryoo** [comment]
@beyondykk9 I've reverted unrelated changes

File: src/parser/semantic_check.c [hunk 3]
```diff
@@ -9709,10 +9709,41 @@ pt_check_grant_revoke (PARSER_CONTEXT * parser, PT_NODE * node)
...
-  /* Replace each Entity Spec with an Equivalent flat list */
-  info.spec_parent = NULL;
-  info.for_update = false;
-  parser_walk_tree (parser, node, pt_flat_spec_pre, &info, pt_continue_walk, NULL);
+  bool is_for_spec = true;
+  PT_NODE *auth_cmd_list = node->info.grant.auth_cmd_list;
+  while (auth_cmd_list)
+    {
+      PT_PRIV_TYPE pt_auth = auth_cmd_list->info.auth_cmd.auth_cmd;
+      if (pt_auth == PT_EXECUTE_PROCEDURE_PRIV || pt_auth == PT_EXECUTE_FUNCTION_PRIV)
```
Comment (ctshim) [inline src/parser/semantic_check.c:9717]: Can it be any other value except PT_EXECUTE_PROCEDURE_PRIV and PT_EXECUTE_FUNCTION_PRIV?

Comment (hgryoo) [reply src/parser/semantic_check.c:9717]: Yes. then `is_for_spec` is not set as false. It means the GRANT/REVOKE statmeent is only for class/vclass

File: src/query/execute_schema.c [hunk 4]
```diff
@@ -1791,24 +1792,52 @@ do_grant (const PARSER_CONTEXT * parser, const PT_NODE * statement)
...
-		  if (error != NO_ERROR)
+		  // TODO: In CBRD-24912, GRANT/REVOKE for stored procedure is implemented, the following will be processed properly
+		  // error = db_grant_object (user_obj, proc_mop)
+		  error = NO_ERROR;	// TODO
+		}
+	    }
+	  else
+	    {
+	      s_list = spec_list;
+	      for (spec = s_list; spec != NULL; spec = spec->next)
```
Comment (beyondykk9) [inline src/query/execute_schema.c]: s_list looks useless.
I suggest "for (spec = spec_list; spec != NULL; spec = spec->next)" without "s_list = spec_list"

**Review by hyunikn — APPROVED** [review]

**Review by beyondykk9 — APPROVED** [review]

**Review by ctshim — APPROVED** [review]
